portfolio/src/main/java/com/google/sps/servlets/DoggoRedditServlet.java

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.FeedPost;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+import java.io.PrintWriter;
+
+@WebServlet("/doggo-servlet")
+public final class DoggoRedditServlet extends HttpServlet {
+
+  static final String COMMENT_TEXT = "commentText";
+  static final String TIMESTAMP = "timestamp";
+  static final String IMAGE_URL = "imageUrl";
+  static final String FEED_POST = "FeedPost";
+  static final String DOG_NAME = "dogName";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Entity postEntity = new Entity(FEED_POST);
+    long timestamp = System.currentTimeMillis();
+    String uploadURL = getUploadedFileUrlFromBlobstore(request, "image");
+
```

Comment:
Consider creating a FeedPost object here and storing that directly in the postEntity, that way you don't have to recreate it in the doGet function

Reply:
Apparently datastore doesn't accept non-primitive data types. However, I converted the FeedPost into a JSON object to store anyway.